opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -59,7 +59,7 @@ public MockTracer(ScopeManager scopeManager, Propagator propagator) {
      * Create a new MockTracer that passes through any calls to inject() and/or extract().
      */
     public MockTracer(Propagator propagator) {
-        this(NoopScopeManager.INSTANCE, propagator);
+        this(new ThreadLocalScopeManager(), propagator);
```

Comment:
can this be done as another PR? It's completely unrelated.

Reply:
Sure, but we'll have to merge that one first.

The MockTracer assumes that `ScopeManager.active()` either returns null or a scope that with a MockSpan. Since we are now returning a NoopScope (and thus a NoopSpan) from the NoopScopeManager, it tries to cast the NoopSpan to a MockSpan and blows up. We won't be able to get the Mock tests to pass until we remove the NoopScopeManager from the Mocks.